fdbserver/OldTLogServer_6_0.actor.cpp

```diff
@@ -1228,21 +1285,173 @@ ACTOR Future<Void> tLogCommit(
 		return Void();
 	}
 
+	// while exec op is being committed, no new transactions will be admitted.
+	// This property is useful for snapshot kind of operations which wants to
+	// take a snap of the disk image at a particular version (no data from
+	// future version to be included)
```

Comment:
It's kind of unhelpful though that enabling and disabling TLog pops will cause a stall in committing... :/

Reply:
enabling and disabling tlog pop should not cause stall in commits, holding the tlog version for the snapshots to complete will cause a stall - this seems unavoidable because we want a set of consistent tlog disk images.  

One possibility is that, we could initiate the snapshot at the tlog after ensuring that the exec version is committed to tlog . Snapshots will get executed in version >= execVersion. In the recovery, we need to instruct the system to recover at execVersion or to recover to the lowest possible version with the knowledge that tlog pops were being disabled at the snapshot create time.

Reply:
> enabling and disabling tlog pop should not cause stall in commits

Wait, then what have I missed?  It looked like you set `execOpCommitInProgress` to `true` for all exec ops, which means that it would be set to true, and thus cause a hiccup in commits, even for enabling and disabling popping?

Reply:
execOpCommitInProgress is set only for snap ops. execVersion is set for all operations, but is unset if it is "tldp" or "tlep", execOpCommitInProgress is set when execVersion is valid. 

I will make the implementation generic - if any command start with \xff then execVersion will not be set and hence execOpCommitInProgress will not be set either. Will edit the comments to reflect that.